### PR TITLE
Allow DEFAULT_LIBRARY_FUNCS_TO_INCLUDE to have native deps. NFC

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #19759

### CODE REVIEW

File: emcc.py [hunk 1]
```diff
@@ -2156,9 +2155,6 @@ def phase_linker_setup(options, state, newargs):
     # See: https://github.com/emscripten-core/emscripten/issues/12065
     # See: https://github.com/emscripten-core/emscripten/issues/12066
     settings.DYNCALLS = 1
-    settings.REQUIRED_EXPORTS += ['emscripten_stack_get_base',
-                                  'emscripten_stack_get_end',
-                                  'emscripten_stack_set_limits']
```
Comment (kripken) [inline emcc.py:2161]: Where can I find the js library deps for this one? (I see the diff in the Asyncify JS library adds one but not all three)

File: emcc.py [hunk 2]
```diff
@@ -2506,10 +2502,6 @@ def phase_linker_setup(options, state, newargs):
       'removeRunDependency',
     ]
 
-  if options.embind_emit_tsd:
-    # TODO: Remove after #19759 is resolved.
-    settings.REQUIRED_EXPORTS += ['free']
```
Comment (kripken) [inline emcc.py:2511]: I'm not sure what this embind option is. Where is the dep for it that replaces this?

**Review by kripken — COMMENTED** [review]
lgtm % questions

File: emcc.py (hunk 2 above)
Comment (sbc100) [reply emcc.py:2511]: I believe its: `src/embind/embind_shared.js:  $getTypeName__deps: ['$readLatin1String', '__getTypeName', 'free'],`

File: emcc.py (hunk 1 above)
Comment (sbc100) [reply emcc.py:2161]: The others are not needed AFAICT.  Maybe they were at some point in the past.

**Pull request merged**